### DISCUSSION TITLE
show image on a url that require auth header

### DISCUSSION THREAD

**Discussion by acaliaro** [skydoves/landscapist#193] [Q&A]
Hi. I try to use Glide with a url that need auth header.
How can I do it?
Thanks

**Comment by faraz152** [comment]
You can pass a `GlideUrl` with `LazyHeaders` as the `imageModel` — this is the standard Glide mechanism for adding custom HTTP headers (auth, cookies, API keys, etc.) and it works directly with landscapist's `GlideImage`.

### Example

```kotlin
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.load.model.LazyHeaders
import com.skydoves.landscapist.glide.GlideImage

val glideUrl = GlideUrl(
    "https://your-api.example.com/protected-image.jpg",
    LazyHeaders.Builder()
        .addHeader("Authorization", "Bearer $yourToken")
        .build()
)

GlideImage(
    imageModel = { glideUrl },
    modifier = Modifier.fillMaxWidth()
)
```

### If headers change at runtime

Wrap it in a `remember` with the token as a key so the URL object is recreated when the token rotates:

```kotlin
val glideUrl = remember(token) {
    GlideUrl(
        imageUrl,
        LazyHeaders.Builder()
            .addHeader("Authorization", "Bearer $token")
            .build()
    )
}

GlideImage(imageModel = { glideUrl })
```

### Multiple headers

`LazyHeaders.Builder` supports chaining multiple `addHeader` calls:

```kotlin
LazyHeaders.Builder()
    .addHeader("Authorization", "Bearer $token")
    .addHeader("X-Client-Id", clientId)
    .build()
```

The key insight is that landscapist's `imageModel` lambda accepts any type that Glide's `ModelLoader` recognises — and `GlideUrl` is Glide's first-class type for URLs with headers.
